File: support_tools/swigwin-3.0.12/Examples/test-suite/go/char_binary_runme.go
package main

import . "./char_binary"

func main() {
	t := NewTest()
	if t.Strlen("hile") != 4 {
		print(t.Strlen("hile"))
		panic("bad multi-arg typemap")
	}

	if t.Strlen("hil\000") != 4 {
		panic("bad multi-arg typemap")
	}

	// creating a raw char*
	pc := New_pchar(5)
	Pchar_setitem(pc, 0, 'h')
	Pchar_setitem(pc, 1, 'o')
	Pchar_setitem(pc, 2, 'l')
	Pchar_setitem(pc, 3, 'a')
	Pchar_setitem(pc, 4, 0)

	Delete_pchar(pc)
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/char_binary_runme.pl
use strict;
use warnings;
use Test::More tests => 10;
BEGIN { use_ok('char_binary') }
require_ok('char_binary');

my $t = char_binary::Test->new();

is($t->strlen('hile'), 4, "string typemap");
is($t->ustrlen('hile'), 4, "unsigned string typemap");

is($t->strlen("hil\0"), 4, "string typemap");
is($t->ustrlen("hil\0"), 4, "unsigned string typemap");

#
# creating a raw char*
#
my $pc = char_binary::new_pchar(5);
char_binary::pchar_setitem($pc, 0, 'h');
char_binary::pchar_setitem($pc, 1, 'o');
char_binary::pchar_setitem($pc, 2, 'l');
char_binary::pchar_setitem($pc, 3, 'a');
char_binary::pchar_setitem($pc, 4, 0);


is($t->strlen($pc), 4, "string typemap");
is($t->ustrlen($pc), 4, "unsigned string typemap");

$char_binary::var_pchar = $pc;
is($char_binary::var_pchar, "hola", "pointer case");

$char_binary::var_namet = $pc;
is($char_binary::var_namet, "hola", "pointer case");

char_binary::delete_pchar($pc);
